Sources/Graphiti/Federation/FederationEntity.swift

```diff
@@ -0,0 +1,11 @@
+import Foundation
+
+public protocol FederationEntityKey: Codable {}
+
+public protocol FederationEntity: Codable {
+    static var typename: String { get }
+}
+
+public extension FederationEntity {
+    static var typename: String { Reflection.name(for: Self.self) }
```

Comment:
I don't think it's compatible with the one in the schema. I just provided a default implementation for simplicity but the user can override it if needed. I did that here since there was a User type already defined in another test. https://github.com/GraphQLSwift/Graphiti/pull/98/files#diff-7086f136baa260fe52dc5fa6c66acf11b265a563f76708d829293b38aa10937cR84